python/ray/_private/worker.py

```diff
@@ -2067,10 +2067,15 @@ def connect(
         # are the same.
         # When using an interactive shell, there is no script directory.
         if not interactive_mode:
-            script_directory = os.path.abspath(os.path.dirname(sys.argv[0]))
-            worker.run_function_on_all_workers(
-                lambda worker_info: sys.path.insert(1, script_directory)
-            )
+            script_directory = os.path.dirname(os.path.realpath(sys.argv[0]))
+            # If driver's sys.path doesn't include the script directory
+            # (e.g driver is started via `python -m`,
+            # see https://peps.python.org/pep-0338/),
+            # then we shouldn't add it to the workers.
+            if script_directory in sys.path:
```

Comment:
If driver's sys.path doesn't include the script directory, do we need to add the current directory into sys.path by run_function_on_all_workers instead of the script directory?

Reply:
The code below adds the current directory into sys.path.